Cabal/src/Distribution/Simple/Hpc.hs

```diff
@@ -68,7 +68,15 @@ mixDir :: FilePath  -- ^ \"dist/\" prefix
        -> Way
        -> FilePath  -- ^ Component name
        -> FilePath  -- ^ Directory containing test suite's .mix files
-mixDir distPref way name = hpcDir distPref way </> "mix" </> name
+mixDir distPref way name = hpcDir distPrefBuild way </> "mix" </> name
+ where
+  -- This is a hack for HPC over test suites, needed to match the directory
+  -- where HPC stores .mix files when the main library of the same package
+  -- is being processed, perhaps in a previous cabal run (#5213).
+  distPrefElements = splitDirectories distPref
+  distPrefBuild = case drop (length distPrefElements - 2) distPrefElements of
```

Comment:
Why `- 2` here? Perhaps it would be worthwhile to include an example of the directory you're trying to access in the comments above, which would make the directory structure more apparent.

Reply:
Will do. The last 2 components are `t/theTestName/`, hence 2.

Reply:
Done.